City-of-Helsinki/drupal-helfi-platform-config
# UHF-1735 Table of contents

How to test with platform:
1. `composer create-project City-of-Helsinki/drupal-helfi-platform:dev-main hel-platform --no-interaction --repository https://repository.drupal.hel.ninja/`
2. `cd hel-platform`
3. `composer require drupal/hdbt:dev-UHF-1735_table_of_contents && composer require drupal/helfi_platform_config:dev-UHF-1735_table_of_contents && composer require drupal/helfi_tpr:dev-UHF-1735_table_of_contents`
4. `make new`
5. Log in to the site using the link provided.
6. Go to add/edit any basic page and select Table of contents checkbox and fill in the title for the table of contents.
7. Create multiple H2 titles with any of the components you can find in Upper content region and Lower content region.
8. Save the page and check that there is Table of contents component between the Lead-in and Upper content elements
Mobile: [Link to design](https://share.goabstract.com/247ac670-4fd6-4279-89a4-05961a53ea6e?collectionLayerId=e9e18de9-38ea-434b-bc10-4ca7e2def100&mode=design&sha=latest)
Desktop: [Link to design](https://share.goabstract.com/247ac670-4fd6-4279-89a4-05961a53ea6e?collectionLayerId=edf55d58-9126-48d5-9099-12f70c578151&mode=design&sha=latest)

How to test with running instance:
- Make sure your instance is up and running
- Update the modules/themes
  - `composer require drupal/hdbt:dev-UHF-1735_table_of_contents && composer require drupal/helfi_platform_config:dev-UHF-1735_table_of_contents && composer require drupal/helfi_tpr:dev-UHF-1735_table_of_contents`
- Run updates: `make drush-cim drush-updb drush-cr`
- Log in if you haven't already `make drush-uli`
- Proceed with item 6 from the list above.
- Also test the Table of contents with Service entity f.e. in Sote "Kiireellinen hoito ja terveyskeskuspäivystys" has plenty of H2 titles already https://helfi-sote.docker.so/fi/sosiaali-ja-terveyspalvelut/admin/content/integrations/tpr-service/8579/edit

Also check these PRs:
https://github.com/City-of-Helsinki/drupal-hdbt/pull/187
https://github.com/City-of-Helsinki/drupal-module-helfi-tpr/pull/73
https://github.com/City-of-Helsinki/drupal-helfi-sote/pull/222
https://github.com/City-of-Helsinki/drupal-helfi-kymp/pull/195
https://github.com/City-of-Helsinki/drupal-helfi-strategia/pull/159
https://github.com/City-of-Helsinki/drupal-helfi-kasvatus-koulutus/pull/12
https://github.com/City-of-Helsinki/drupal-helfi-tyo-yrittaminen/pull/8
